daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DarReader.scala

```diff
@@ -94,7 +96,6 @@ object GenDarReader {
   }
 }
 
-object DarReader extends GenDarReader[ArchivePayload](is => Try(Reader.readArchive(is)))
-
-object RawDarReader
-    extends GenDarReader[DamlLf.Archive](is => Try(DamlLf.Archive.parseFrom(is.toByteString)))
+object DarParser extends GenDarReader[DamlLf.Archive](Decoder.ArchiveParser)
+object DarReader extends GenDarReader[ArchivePayload](Decoder.ArchiveReader)
+object DarDecoder extends GenDarReader[(PackageId, Ast.Package)](Decoder.ArchiveDecoder)
```

Comment:
Would these names make it easier to remember which is which?
```suggestion
object DarArchiveReader extends GenDarReader[DamlLf.Archive](Decoder.ArchiveParser)
object DarArchivePayloadReader extends GenDarReader[ArchivePayload](Decoder.ArchiveReader)
object DarPackageAstReader extends GenDarReader[(PackageId, Ast.Package)](Decoder.ArchiveDecoder)
```

Reply:
The convention is as follow:
- parsing = proto parsing (i.e. bytes to proto)
- reading = proto parsing + validation (hash and version)
- decoding = proto parsin + validation + proto decoding (proto -> Ast)